docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+* // how it interacts with the other components
+
+/* TODO explain the various variables and methods */
+
+### 3.3. Commands Component (Neil)
+
+![UML diagram for Timetable Commands](/diagrams/class/jpeg/timetable_commands.jpg)
+
+![UML diagrams for Notebook Commands](/diagrams/class/jpeg/notebook_commands.jpg)
+
+/* to insert UML diagram */
+
+The `Commands` component,
+
+// TODO add components
+
+### 3.4. Tasks Component
+
+/* to insert UML diagram */
+
+### 3.5 Notebooks Component
+
+![UML diagram for Notebooks](/diagrams/class/jpeg/notebooks.jpg)
+
+/* TODO explain the various variables and methods */
+
+### 3.6. Storage Component
+
+/* to insert UML diagram */
+
+The `Storage` component,
+
+* can save the name of user in a .txt file and read it back.
+
+* can save the Tasklist and NotebookShelf data in a .txt file and read it back.
+
+## 4. Implementation
+
+The following section describes the implementation of certain key features in the current version of Zer0Note. It also
+provides some background into our (the original developers of Zer0Note) thinking and the rationale behind the
+decisions.
+
+### 4.1. Mode Switch Feature
+
+#### 4.1.1. Implementation
+
+/* work in progress */
+
+#### 4.1.2. Design Considerations
+
+/* work in progress */
+
+### 4.2. Timetable Mode
+
+#### 4.2.1. TaskList Management Feature
+
+##### 4.2.1.1. Implementation
+
+`TaskList` is implemented to manage and store the tasks input by the user. It comprises an `ArrayList` list of
+ `Task`s, and a few helper methods.
+
+This means that multiple operations such as addition and deletion can be done on a `Task`, without affecting
+the contents of other `Task` in the `TaskList`.
+<br>
+
+Given below is an example usage scenario and how the add task function behaves.
+
+Step 1. The user launches the application for the first time. CliUserInterface#executeCommand is called when the user adds a `Task` into the `TaskList`.
+
+Step 2. The user types `add /tTask /by19-10-2020 1900`. The `add` command is passed through `InputParser#getCommandFromInput`, which then creates a constructor for `AddCommandTimetableMode`.
+
+Step 3. `AddCommandTimetableMode#execute()` is called, which then calls `InputParser#parseTaskTitle`, which first extracts the `title` from the user's input.
+
+Step 4. `InputParser#parseDeadline` is then called, which returns the `deadline` to `AddCommandTimetableMode#execute()`.
+
+Step 5. `TaskList#addTask` is then called and a new `Task`, with `title` and `deadline`, is initialised.
+
+Step 6. To signal that the user has successfully added a task, a message is printed with `CliMessages#printAddedTaskMessage`.
+
+The UML sequence diagram below shows how the add task command works.
+![Sequence Diagram for Add Task Command](/diagrams/class/jpeg/add_task.jpg)
+
+<hr>
+
+The figure below shows how the delete task command works:
+<img src= "https://user-images.githubusercontent.com/60319628/96657942-02dc6900-1376-11eb-9284-38322e1a2b09.png">
```

Comment:
For this UML diagram, for the methods that execute within a class like "executeCommand(userInput)", isn't there supposed to be another box overlaid on top of the existing box? Like how it's shown for the executeCommand in the tag operation sequence diagram?